### PR TITLE
GitHub Open Source Community Standards

### LINKED ISSUE #9: GitHub Open Source community standards
Adapt to the [recommended community standards](https://github.com/blog/2380-new-community-tools) using the [Open Source Guide](https://opensource.guide) to complete the projects [community profile](https://github.com/communicode/communikey-backend/community).

### [Code of Conduct](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/)

It should include sections about

* [x] what we believe in and how we act
* [x] unacceptable behavior
* [x] the responsibilities of the maintainer
* [x] the enforcement of the Code of Conduct
* [x] consequences for violations
* [x] the scope of the Code of Conduct

### [Contributing Guidelines](https://github.com/blog/1184-contributing-guidelines)

It should include sections about

* [x] how to get started
* [x] bug reports
* [x] enhancement suggestions
* [x] pull requests
* [x] styleguides
  *  [x] Java Code Style
  * [x] [Git Commit Messages](https://github.com/arcticicestudio/styleguide-git)
* [x] credits

### PR DESCRIPTION
**Pull request by svengreb**
> Closes #9 
Blocks #19 (must be merged **before**)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,92 +1,205 @@
...
-In the interest of fostering an open and welcoming environment, we as
-contributors and maintainers pledge to making participation in our project and
-our community a harassment-free experience for everyone, regardless of age, body
-size, disability, ethnicity, gender identity and expression, level of experience,
-nationality, personal appearance, race, religion, or sexual identity and
-orientation.
+The project development workflow and process uses [GitHub Issues][gh-issues]- and [Pull Requests][gh-pr] management to track issues and pull requests.
 
-### Our Standards
+Before you continue with this contribution guideslines we highly recommend to read the awesome GitHub [Open Source Guide](https://opensource.guide) on how to [making open source contributions][gh-osguide-contribute].
```
Comment (fermentfan) [inline CONTRIBUTING.md]: > Before you continue with **these** contribution **guidelines**

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Code of Conduct
+
+All members, committers and volunteers in this project are required to act according to the following Code of Conduct. We ask you to follow these guidelines which help steer our interactions and strive to keep communikey a positive and growing project and help us provide and ensure a safe environment for everyone.
+
+If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact one of the core team members via email. Your reports will be taken seriously and not dismissed or argued with.
+
+## What we believe in and how we act
+
+* In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation. We welcome people regardless of these or other attributes.
```
Comment (fermentfan) [inline CODE_OF_CONDUCT.md]: >  we as contributors and maintainers pledge to **make** participation

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* Publishing others' private information, such as a physical or electronic address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a professional setting
+
+## Responsibilities
+
+The members of the core team and project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project core team members or owner via [email][email]. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
Comment (fermentfan) [inline CODE_OF_CONDUCT.md]: Add our communicode email addresses?

**Review by fermentfan — CHANGES_REQUESTED** [review]
Fix grammar

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (svengreb) [reply CODE_OF_CONDUCT.md]: The main idea of this (broken) link was to add a support email address like e.g. communikey-support@communicode.de, but I think we should start by adding a [`.mailmap`][git-mm] file and link to this instead. This prevents unnecessary overhead of updating all documents when new team members are added and additionally adds the main functionality of the file: Mapping commits when someone uses a different email address.

[git-mm]: https://git-scm.com/docs/git-shortlog#_mapping_authors

**Review by fermentfan — APPROVED** [review]

**Review by lleifermann — APPROVED** [review]

**Pull request merged**